sphinx/application/data/features/feature-repository/src/main/java/chat/sphinx/feature_repository/SphinxRepository.kt

```diff
@@ -3822,6 +3826,62 @@ abstract class SphinxRepository(
         return response ?: Response.Error(ResponseError("Returned before completing"))
     }
 
+    override suspend fun saveProfile(
+        relayUrl: String,
+        host: String,
+        key: String
+    ): Response<Boolean, ResponseError> {
+        var response: Response<Boolean, ResponseError>? = null
+
+        applicationScope.launch(mainImmediate) {
+            networkQuerySaveProfile.getProfileByKey(host, key).collect { loadResponse ->
+                when (loadResponse) {
+                    is LoadResponse.Loading -> {}
+
+                    is Response.Error -> {
+                        response = loadResponse
+                    }
+
+                    is Response.Success -> {
+                        val body = moshi.adapter(PersonInfoDto::class.java).fromJson(loadResponse.value.body)
+                        if(body != null) {
+                            val profileInfo = PersonInfoDto(
+                                id = body.id,
+                                host = body.host,
+                                owner_alias = body.owner_alias,
+                                description = body.description,
+                                img = body.img,
+                                tag = body.tag,
+                                price_to_meet = body.price_to_meet,
+                                extras = body.extras
+                            )
```

Comment:
You can use the PersonInfoDto data class on the SaveProfileDto directly so you don't need to parse it with moshi and then create a new object to send on the saveProfile call. We usually do not do it this way, but creating Dtos for the responses to be parsed automatically

Reply:
I made changes in 7023316 for this but I'm still using moshi `loadResponse.value.body` comes as a string with all the `SavePeopleProfileDto` in the string is there a better way of making that string into an `SavePeopleProfileDto` object without moshi?

Reply:
If you set the type of the body as PersonInfoDto on the SavePeopleProfileDto it should be parsed automatically, then you can use the loadResponse.value.body (which will be of type PersonInfoDto) to be sent on the /profile endpoint. It needs to be tested but it should work as it works on ContactDto where the invite key contains an InviteDto object.